### ISSUE TITLE
Web Ui Depend on Manga.Infrastructure

### ISSUE THREAD

**Issue by pouyaSamie** [ivanpaulovich/clean-architecture-manga#86]
Hello and thank you for your great project.
i have a question why we have Infrastructure reference in Web Ui ?
base on diagrams we should not depend on Infrastructure ?
so if we have to add the reference because of registering DI classes is there any best practice for loos that dependency?

**Comment by ivanpaulovich** [comment]
Hi @pouyaSamie,

The application entry-point should know the abstractions and their implementations:

![Layered app](https://paulovich.net/img/Untitled-Diagram-1.png)

If you see the Manga source code, the only code that uses the infrastructure classes is at the startup in order to builder the Container.

If you wish, you have options to decouple this:

1) Load the infrastructure dynamically using Autofac:

https://github.com/ivanpaulovich/event-sourcing-jambo/blob/master/source/Producer/Jambo.Producer.UI/autofac.json

2) Make the infrastructure the application entry-point, which is very exotic in my opinion:

https://github.com/ivanpaulovich/mfa/tree/master/source/Jambo.Producer.Infrastructure

The bottom line is that most developers are used to see the UI referencing all other dependencies, to make it in a different way you will need to teach everyone about this decision.

**Comment by ivanpaulovich** [comment]
Closing it. You are welcome to reopen in case of new issues.
